packages/calypso-e2e/src/element-helper.ts

```diff
@@ -152,3 +152,28 @@ export async function getIdFromBlock( block: Locator ): Promise< string > {
 
 	return blockId;
 }
+
+/**
+ * Resolves once widgets.wp.com "message" events become idle. This can be
+ * synonymous to the moment all the widgets become ready to use / stable. This
+ * helper can be used, e.g., on Atomic pages where the widgets are loaded via
+ * iframes causing layout shifting, and in consequence Playwright's stability
+ * checks to fail.
+ *
+ * @param {Page} page The parent page object.
+ */
+export async function waitForWPWidgets( page: Page ): Promise< void > {
+	await page.evaluate( async () => {
+		await new Promise( ( resolve ) => {
+			const wait = () => setTimeout( resolve, 1000 );
+			let timeout = wait();
+
+			window.addEventListener( 'message', ( event ) => {
+				if ( event.origin === 'https://widgets.wp.com' ) {
+					clearTimeout( timeout );
+					timeout = setTimeout( resolve, 1000 );
```

Comment:
**Suggestion (non-blocking)**: (Mostly for consistency), I believe you can re-use that `wait` function you've defined above here?

Reply:
Updated in https://github.com/Automattic/wp-calypso/pull/69180/commits/13a2017cc15a5e39fe5f64093e97406268faf21b